File: java_src/limelight/Page.java
package limelight;

import java.util.Hashtable;

public class Page extends Block
{
	private Hashtable<String, Style> styles;

	public Page()
	{
		super();
		setPage(this);
		styles = new Hashtable<String, Style>();
	}

	public Hashtable<String, Style> getStyles()
	{
		return styles;
	}
}

File: ruby_src/lib/limelight/page.rb
require 'limelight/java_util'
require 'limelight/block'
require 'limelight/button_group_cache'

module Limelight
  class Page < Block
    
    include Java::limelight.ui.Page
  
    attr_reader :button_groups, :styles, :illuminator
    attr_accessor :book, :loader, :visible
    getters :book, :loader
    setters :book
    
    def initialize(options={})
      super(options)
      @page = self
      @button_groups = ButtonGroupCache.new
      illuminate
    end
  
    def add_options(options)
      @options = options
      illuminate
    end
    
    def illuminate
      @styles = @options.has_key?(:styles) ? @options.delete(:styles) : (@styles || {})
      @illuminator = @options.delete(:illuminator) if @options.has_key?(:illuminator)
      super
    end
  
  end
end